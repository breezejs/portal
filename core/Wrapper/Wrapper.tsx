import React, {memo, ReactNode} from 'react';
import styled from 'styled-components';
import {GlobalStylesTheme} from '../GlobalStyles';

const Main = styled.main`
  background-color: ${GlobalStylesTheme.bodyBg};
  padding: ${GlobalStylesTheme.gridGutter};
`;

interface IWrapperProps {
  children?: ReactNode;
}

function Wrapper ({children}: IWrapperProps) {
  return (
    <Main>{children}</Main>
  );
}

export default memo(Wrapper);
