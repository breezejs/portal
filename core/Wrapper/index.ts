export {default as default} from './Wrapper';
