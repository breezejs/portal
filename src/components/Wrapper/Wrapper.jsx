import PropTypes from 'prop-types';
import React, {memo} from 'react';
import styled from 'styled-components';
import {GlobalStylesTheme} from '../GlobalStyles';

const Main = styled.main`
  background-color: ${GlobalStylesTheme.bodyBg};
  padding: ${GlobalStylesTheme.gridGutter};
`;

function Wrapper ({children}) {
  return (
    <Main>{children}</Main>
  );
}

Wrapper.propTypes = {
  children: PropTypes.node.isRequired
};

export default memo(Wrapper);
