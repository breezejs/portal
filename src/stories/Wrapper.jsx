import {text, withKnobs} from '@storybook/addon-knobs';
import {storiesOf} from '@storybook/react';
import React from 'react';
import GlobalStyles from '../components/GlobalStyles';
import Wrapper from '../components/Wrapper';

const stories = storiesOf('Wrapper', module);

stories
  .addDecorator(withKnobs)
  .addDecorator((storyFn) => (
    <div>
      <GlobalStyles />
      {storyFn()}
    </div>
  ))
  .add('with content', () => (
    <Wrapper>{text('Content', 'This component will wrap its content within it')}</Wrapper>
  ));
