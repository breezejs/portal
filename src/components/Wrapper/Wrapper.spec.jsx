import {mount, shallow} from 'enzyme';
import React from 'react';
import Wrapper from './Wrapper';

describe('Wrapper', () => {
  it('should render', () => {
    const result = shallow(
      <Wrapper>Content</Wrapper>
    );

    expect(result).toMatchSnapshot();
  });

  it('should have the correct css styles', () => {
    const result = mount(
      <Wrapper>Content</Wrapper>
    );

    expect(result.find('main')).toMatchSnapshot();
  });
});
