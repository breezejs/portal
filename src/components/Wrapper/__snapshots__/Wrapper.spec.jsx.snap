// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper should have the correct css styles 1`] = `
.c0 {
  background-color: #fff;
  padding: 1rem;
}

<main
  className="c0"
>
  Content
</main>
`;

exports[`Wrapper should render 1`] = `
<styled.main>
  Content
</styled.main>
`;
